### PR TITLE
Add chartconfig for chart-operator

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#1902

Adds the chartconfig CRD.

### CODE REVIEW

File: pkg/apis/core/v1alpha1/chart_types.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+//     metadata:
+//       name: certconfigs.core.giantswarm.io
+//     spec:
+//       group: core.giantswarm.io
+//       scope: Namespaced
+//       version: v1alpha1
+//       names:
+//         kind: CertConfig
+//         plural: certconfigs
+//         singular: certconfig
```
Comment (xh3b4sd) [inline pkg/apis/core/v1alpha1/chart_types.go]: Let's please get the comments straight.

File: pkg/apis/core/v1alpha1/chart_types.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+type ChartConfigSpec struct {
+	Chart         ChartConfigSpecChart         `json:"chart" yaml:"chart"`
+	VersionBundle ChartConfigSpecVersionBundle `json:"versionBundle" yaml:"versionBundle"`
+}
+
+type ChartConfigSpecChart struct {
+	Channel string `json:"channel" yaml:"channel"`
+	Name    string `json:"name" yaml:"name"`
```
Comment (xh3b4sd) [inline pkg/apis/core/v1alpha1/chart_types.go]: I am not into details. What is name about? Why not ID? Maybe some godoc with examples would be useful as well.

File: pkg/apis/core/v1alpha1/chart_types.go (hunk 1 above)
Comment (rossf7) [reply pkg/apis/core/v1alpha1/chart_types.go]: Fixed.

File: pkg/apis/core/v1alpha1/chart_types.go (hunk 2 above)
Comment (rossf7) [reply pkg/apis/core/v1alpha1/chart_types.go]: Name is the fully qualified name of the Helm chart. Channel is the release channel to reconcile against. Good idea I've added godoc and examples.

File: pkg/apis/core/v1alpha1/chart_types.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+type ChartConfigSpecChart struct {
+	// Channel is the name of the Appr channel to reconcile against.
+	// e.g. 1.0-stable
+	Channel string `json:"channel" yaml:"channel"`
+	// Name is the fully qualified name of the Helm chart to deploy.
+	// e.g. quay.io/giantswarm/chart-operator-chart
+	Name string `json:"name" yaml:"name"`
+}
```
Comment (xh3b4sd) [inline pkg/apis/core/v1alpha1/chart_types.go]: Nice. Thanks for the comments.

**Pull request merged**